### PR TITLE
Implement SqlOfTContext

### PR DESCRIPTION
**Pull request by zpqrtbnk**

### CODE REVIEW

File: src/NPoco/Sql.cs [hunk 1]
```diff
@@ -124,43 +120,66 @@ private void Build(StringBuilder sb, List<object> args, Sql lhs)
...
         {
-            return new SqlJoinClause(Append(new Sql(JoinType + table)));
+            Append(joinType + table);
+            return new SqlJoinClause(this);
         }
 
         public SqlJoinClause InnerJoin(string table) { return Join("INNER JOIN ", table); }
         public SqlJoinClause LeftJoin(string table) { return Join("LEFT JOIN ", table); }
+        public SqlJoinClause RightJoin(string table) { return Join("RIGHT JOIN ", table); }
+        public SqlJoinClause LeftOuterJoin(string table) { return Join("OUTER LEFT JOIN ", table); }
```
Comment (schotime) [inline src/NPoco/Sql.cs]: left outer join is the same as left join im pretty sure

Comment (zpqrtbnk) [inline src/NPoco/Sql.cs]: silly me - yet I _think_ I was trying to do something here - let me check

Comment (zpqrtbnk) [inline src/NPoco/Sql.cs]: fixed

**Comment by schotime** [comment]
Do you want to clean this PR up by removing unnecessary comments and squashing the two commits so that I can merge in soon. ;)

**Comment by zpqrtbnk** [comment]
Sure, asap.

**Comment by zpqrtbnk** [comment]
Done, now should be one clean commit.

**Comment by zpqrtbnk** [comment]
Anything else you want me to do?

**Comment by schotime** [comment]
Don't think so. Will try and get it into next release this week. Apologies for the delay.

**Comment by schotime** [comment]
Merged and is in 3.3.0-beta3.
Let me know how it goes and then it should be good for a release

**Comment by zpqrtbnk** [comment]
Upgraded our dev branch to NPoco 3.3.0-beta3 and all seems OK. Our custom branch now only has a few cosmetic diffs that I'll eventually PR to you, but it looks like we are now running on stock NPoco. Woot!!

**Pull request closed without merging**